File: 14Jan2024.java
import java.util.Arrays;

class Solution {
    public boolean closeStrings(String word1, String word2) {
        if (word1.length() != word2.length())
            return false;
        int[] fre1 = new int[26];
        int[] fre2 = new int[26];
        for (int i = 0; i < word1.length(); i++) {
            fre1[word1.charAt(i) - 'a']++;
            fre2[word2.charAt(i) - 'a']++;
        }

        for (int i = 0; i < 26; i++) {
            if (fre1[i] != fre2[i]) {
                if (fre1[i] == 0 || fre2[i] == 0)
                    return false;
            } else {
                fre1[i] = fre2[i] = -1;
            }
        }

        Arrays.sort(fre1);
        Arrays.sort(fre2);
        int ct = 0;
        for (int i = 0; i < 26; i++) {
            if (fre1[i] != fre2[i]) {
                return false;
            } else {
                ct++;
            }
        }
        if (ct % 2 == 1)
            return false;
        return true;

    }
}

File: 14Jan2024.py
class Solution:
    def closeStrings(self, w1: str, w2: str) -> bool:
        if(len(w1)!=len(w2)) :
            return False
        f1 = [0]*26
        f2 = [0]*26
        for i in range(0,len(w1)) :
            f1[ord(w1[i])-ord('a')]+=1
            f2[ord(w2[i])-ord('a')]+=1
        
        for i in range(0,26) :
            if(f1[i]==f2[i]) :
                f1[i]=f2[i]=-1
            else :
                if(f1[i]==0 or f2[i]==0) :
                    return False
        
        f1.sort()
        f2.sort()
        ct=0
        for i in range(0,26) :
            if(f1[i]!=f2[i]) :
                return False
            else :
                ct+=1
        
        if(ct%2==1) :
            return False
        return True